### ISSUE TITLE
Spack Broken -  ImportError: No module named ordereddict

### ISSUE THREAD

**Issue by JusticeForMikeBrown** [spack/spack#6929]
Hi

I pulled latest from ```develop``` and now I cannot run Spack:

```
sysadmin@node118[~/repos/spack]$ git log | head -n 20
commit 9e3f8c0d79d6f2c02f63599cf58527d596e1a445
Author: 健美猫 <weijianwen@gmail.com>
Date:   Fri Jan 12 21:50:07 2018 +0800

    A new package: ethminer. (#6908)

commit 03ff2a32aef078f7b091d0a8da1e6bb2a1491081
Author: Adam J. Stewart <ajstewart426@gmail.com>
Date:   Thu Jan 11 16:14:26 2018 -0600

    Reduce verbosity of build tests (#6901)

commit 070bcab7f2655b110447bb3b97e8733d98742e74
Author: Michael Kuhn <suraia@ikkoku.de>
Date:   Thu Jan 11 17:19:44 2018 +0100

    libbson, libmongoc: Update to 1.9.1 (#6900)

commit 5dd7baa61db49df86190f5773e1153e4f458125b
Author: stefanosoffia <stefanosoffia@users.noreply.github.com>

```

```
sysadmin@node118[~/repos/spack]$ git status .
# On branch develop
nothing to commit (working directory clean)
```

```
sysadmin@node118[~/repos/spack]$ spack
Traceback (most recent call last):
  File "/home/sysadmin/repos/spack/bin/spack", line 81, in <module>
    sys.exit(spack.main.main())
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/spack/main.py", line 502, in main
    parser.print_help()
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/external/argparse.py", line 2360, in print_help
    self._print_message(self.format_help(), file)
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/spack/main.py", line 264, in format_help
    return self.format_help_sections(level)
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/spack/main.py", line 141, in format_help_sections
    add_all_commands(self)
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/spack/main.py", line 103, in add_all_commands
    parser.add_command(cmd)
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/spack/main.py", line 253, in add_command
    module = spack.cmd.get_module(name)
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/spack/cmd/__init__.py", line 89, in get_module
    level=0)
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/spack/cmd/test.py", line 31, in <module>
    import pytest
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/external/pytest.py", line 13, in <module>
    from _pytest.fixtures import fixture, yield_fixture
  File "/pbtech_mounts/homes027/sysadmin/repos/spack/lib/spack/external/_pytest/fixtures.py", line 25, in <module>
    from ordereddict import OrderedDict
ImportError: No module named ordereddict
```

**Comment by alalazo** [comment]
@adamjstewart We upgraded `pytest` lately, did we forget to vendor some new dependency? In `_pytest/fixtures.py` I see:
```python
if sys.version_info[:2] == (2, 6):
    from ordereddict import OrderedDict
else:
    from collections import OrderedDict  # nopyqver
```
By the way: `ordereddict` is a very unfortunate one, as we have our own version of it for Spack.

**Comment by alalazo** [comment]
@JusticeForMikeBrown Can you check if reverting 57c71aea8943b1c0f8b4fd1b985911d2af6d4dca it works for you?

**Comment by adamjstewart** [comment]
Oddly enough, with the system Python 2.6.6 on CentOS 6, `spack` works just fine. I can even run `spack test`.
```
$ python
Python 2.6.6 (r266:84292, Aug 18 2016, 15:13:37) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-17)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import ordereddict
```
Perhaps it works for some versions of Python 2.6 and not others? That would explain why the Travis 2.6 tests passed.

**Comment by adamjstewart** [comment]
#6930 implements suggestion 3, although I may also take a look to see how difficult 1 would be.

**Comment by alalazo** [comment]
> I don't think we need a new dependency, just need to update that reference

Option 4. like @adamjstewart 1. but we vendor `ordereddict` only for python 2.6, and use `collections.OrderedDict` in other cases. If you want I can put this together cherry-picking it from a branch of mine.

**Comment by adamjstewart** [comment]
@alalazo I like option 4. Go for it.

**Comment by JusticeForMikeBrown** [comment]
Hi

Sorry I couldn't really help.  Swamped at work and now heading out of town!

**Comment by alalazo** [comment]
@JusticeForMikeBrown The issue should have been solved by #6931. If not, please reopen this.

**Comment by JusticeForMikeBrown** [comment]
Thanks!
